factories/GenericPromo.php

```diff
@@ -24,7 +24,7 @@ public function create($limit = 1, $flatten = false, $options = [])
     {
         $promo_group_id = $this->faker->randomNumber(5);
 
-        for ($i = 1; $i <= $limit; $i++) {
+        for ($i = 0; $i <= ($limit - 1); $i++) {
```

Comment:
Small thing here, this could be done without the internal parenthesis:
`for ($i = 0; $i < $limit; $i++) {`